index.js

```diff
@@ -1,68 +1,65 @@
+import { monsters } from './monsters.js';
 
-import {monsters} from './monsters.js';
-console.log(monsters);
-for(let monsterDiv of monsters){
-    showMonster(monsterDiv);
+const monsterContainer = document.querySelector('.monsters');
+const monsterFilter = document.querySelector('#search-monster');
 
+function populateMonsters(content) {
+  // Clear out existing content
+  monsterContainer.innerHTML = '';
+  // Usde showMonster to generate an Array of DOMNodes
+  let createMonsters = content.map((el) => showMonster(el));
+  // Append the DOMNodes in one DOM transaction
+  monsterContainer.append(...createMonsters);
 }
```

Comment:
The purpose of the populateMonsters() function is to accept an Array of monster objects, clear out the container, generate an Array of DOM nodes from the data and then inject the created nodes in one DOM transaction. This improves efficiency.